doc/compiler/control/CompilationControl.md

```diff
@@ -0,0 +1,304 @@
+decreasing addresses.
+
+### Code Cache Reclamation
+
+When a method is recompiled, the old body eventually stops being used,
+which is a waste of space in the Code Cache. In order to maximize the
+usable space in the Code Cache, the JIT reclaims stale bodies; see the
+[Code Cache Reclamation doc](../runtime/CodeCacheReclamation.md) for
+more details.
+
+
+# Influencing Factors
+
+There are many factors that influence the decisions the Control
+Infrastructure makes. The JIT has to balance improving application
+performance with not taking away too many resources from the
+application. The following are some of the considerations that have
+the biggest impact and influence.
+
+## JIT State
+
+The JIT maintains several states that reflect the various execution
+characteristics of an application. The Compilation Control heuristics
+depend on the state, or phase, the JIT is in. The JIT transitions
+between the following states:
+
+- IDLE_STATE
+  - Decisions focus on minimizing Compilation Thread CPU usage.
+- STARTUP_STATE
+  - Decisions focus on making Startup as fast as possible.
+- RAMPUP_STATE
+  - Decisions focus on making Rampup as fast as possible.
+- STEADY_STATE
+  - Decisions focus on maintaining throughput.
+- DEEPSTEADY_STATE
+  - Decisions focus on maintaining throughput while minimizing CPU
+    consumed by the JIT (including the Sampler Thread).
+
+## Available Resources
+
+The JIT should minimize its impact on application performance.
+Therefore, how aggressively it can compile methods depends on
+the resources available to the JVM. Multiple compilation threads may
+only be activated if the system is not resource constrained.
+To minimize memory usage, only one higher optimization level
+(hot, very-hot, scorching) compile is performed at a time.
+Additionally, the maximum amount of memory available for a compilation
+depends on the amount of physical memory available.
+
+## JIT Hooks
+
+The JIT registers several hooks with the VM. These hooks are invoked
+when the environment changes at well defined points. These changes
+include Class Loading / Unloading, Code Redefinition, Garbage
+Collection, etc. These events can cause invalidations both in the
+generated code and the compilation. The former is addressed using
+[Runtime Assumptions](../runtime/RuntimeAssumption.md); the latter
+is addressed by aborting the compilation and possibly reattempting
+it.
+
+## Shared Classes Cache
+
+The existence of the Shared Classes Cache (SCC) has a sigificant
+influence on the heuristics; this is because of SCC enables
+Ahead of Time (AOT) Compilation. See the [AOT docs](../aot) for
+technical details on the AOT feature.
+
+Because AOT compilations enable very fast startup, the JIT tries
+to maximize the number of compiled bodies it can load from the SCC
+(referred to as AOT Load). As a consequence, the invocation counts
+for AOT Compilations are lower than JIT Compilations. Additionally,
```

Comment:
> As a consequence, the invocation counts for AOT Compilations are lower than JIT Compilations

This is not clear. It's the method that receives an invocation count, not the compilation. Moreover, we cannot know precisely which method will be AOTed.